Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/pull_request.py

```diff
@@ -272,9 +273,27 @@ def pull_request_branch_point(cls, repository, args, **kwargs):
             ):
                 sys.stderr.write("Abandoning pushing pull-request because '{}' could not be created\n".format(args.issue))
                 return None
+
         elif args.issue and repository.branch != args.issue:
-            sys.stderr.write("Creating a pull-request for '{}' but we're on '{}'\n".format(args.issue, repository.branch))
-            return None
+            error = "Creating a pull-request for '{}' but we're on '{}'\n".format(args.issue, repository.branch)
+            if not repository.dev_branches.match(repository.branch):
+                sys.stderr.write(error)
+                return None
+
+            if string_utils.decode(args.issue).isnumeric():
+                issue = Tracker.instance().issue(int(args.issue))
+            else:
+                issue = Tracker.from_string(args.issue)
+            if not issue:
+                sys.stderr.write(error)
+                return None
+            if not repository.branch.endswith('/{}'.format(issue.id)) and not repository.branch.endswith('/{}'.format(Branch.to_branch_name(issue.title))):
+                sys.stderr.write(error)
+                return None
```

Comment:
I don't feel great about branch names being magical in this way. On one hand, git-webkit shouldn't care what my branch is, and I should be able to name branches freely. But on the other hand, anyone using `git webkit pr --issue` is already giving the tool control over their branch name.

Reply:
The other option here is that we tie a branch name to a specific bug via `git config`. We started down this path already, I didn't use that technique here because the user is likely to interrupt the script before we tied branch name and specific bug together. Open to changing things in the future, but as you point out, `--issue` already implies the user wishes to create a branch.